src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -262,64 +280,70 @@ p5.RendererGL.prototype._calculateEdges = function(
-      for (i = 0; i < verts.length - 2; i = i + 3) {
-        res.push([i, i + 1]);
-        res.push([i + 1, i + 2]);
-        res.push([i + 2, i]);
-      }
-      break;
-    case constants.LINES:
-      for (i = 0; i < verts.length - 1; i = i + 2) {
-        res.push([i, i + 1]);
-      }
-      break;
-    case constants.QUADS:
-      // Quads have been broken up into two triangles by `vertex()`:
-      // 0   3--5
-      // | \  \ |
-      // 1--2   4
-      for (i = 0; i < verts.length - 5; i += 6) {
-        res.push([i, i + 1]);
-        res.push([i + 1, i + 2]);
-        res.push([i + 3, i + 5]);
-        res.push([i + 4, i + 5]);
-      }
-      break;
-    case constants.QUAD_STRIP:
-      // 0---2---4
-      // |   |   |
-      // 1---3---5
-      for (i = 0; i < verts.length - 2; i += 2) {
-        res.push([i, i + 1]);
-        res.push([i, i + 2]);
-        res.push([i + 1, i + 3]);
-      }
-      res.push([i, i + 1]);
-      break;
-    default:
-      for (i = 0; i < verts.length - 1; i++) {
-        res.push([i, i + 1]);
-      }
-      break;
-  }
-  if (shouldClose) {
-    res.push([verts.length - 1, 0]);
+  let j = 0;
+  while (j < 2) {
+    let part = [];
+    switch (shapeMode) {
+      case constants.TRIANGLE_STRIP:
+        for (i = 0; i < verts[j].length - 2; i++) {
+          part.push([i, i + 1]);
+          part.push([i, i + 2]);
+        }
+        part.push([i, i + 1]);
+        break;
+      case constants.TRIANGLE_FAN:
+        for (i = 1; i < verts[j].length - 1; i++) {
+          part.push([0, i]);
+          part.push([i, i + 1]);
+        }
+        part.push([0, verts[j].length - 1]);
+        break;
+      case constants.TRIANGLES:
+        for (i = 0; i < verts[j].length - 2; i = i + 3) {
+          part.push([i, i + 1]);
+          part.push([i + 1, i + 2]);
+          part.push([i + 2, i]);
+        }
+        break;
+      case constants.LINES:
+        for (i = 0; i < verts[j].length - 1; i = i + 2) {
+          part.push([i, i + 1]);
+        }
+        break;
+      case constants.QUADS:
+        // Quads have been broken up into two triangles by `vertex()`:
+        // 0   3--5
+        // | \  \ |
+        // 1--2   4
+        for (i = 0; i < verts[j].length - 5; i += 6) {
+          part.push([i, i + 1]);
+          part.push([i + 1, i + 2]);
+          part.push([i + 3, i + 5]);
+          part.push([i + 4, i + 5]);
+        }
+        break;
+      case constants.QUAD_STRIP:
+        // 0---2---4
+        // |   |   |
+        // 1---3---5
+        for (i = 0; i < verts[j].length - 2; i += 2) {
+          part.push([i, i + 1]);
+          part.push([i, i + 2]);
+          part.push([i + 1, i + 3]);
+        }
+        part.push([i, i + 1]);
+        break;
+      default:
+        for (i = 0; i < verts[j].length - 1; i++) {
+          part.push([i, i + 1]);
+        }
+        break;
+    }
+    if (shouldClose) {
+      part.push([verts[j].length - 1, 0]);
+    }
+    res.push(part);
```

Comment:
On the main branch, the expected output of this function is an array of `[index1, index2]` tuples, basically saying "connect the vertex at `geometry.vertices[index1]` with the vertex at `geometry.vertices[index2]` with a line." This starts to get a little tricky now that not all the vertices are in the same `geometry.vertices` array.

With your change, this now returns an array of arrays of those tuples. That could be helpful, since now we know that the first array element is the pairs for the main shape vertices, the second array element is the pairs for the first contour, the third array element is the pairs for the second contour, etc. It just means we'll have to update [the `_edgesToVertices` function](https://github.com/processing/p5.js/blob/393ddbc50457734cba8b2128aa20ec43fc62de7b/src/webgl/p5.Geometry.js#L267), which takes this output and makes all the lines, to handle this new format.

e.g. over here, it directly accesses `this.vertices[index]`. We'll have to update this to loop over each array of edges, and determine which set of vertices/vertex stroke colors it should be reading from: https://github.com/processing/p5.js/blob/393ddbc50457734cba8b2128aa20ec43fc62de7b/src/webgl/p5.Geometry.js#L281